src/gp/src/GPMSA.C

```diff
@@ -847,6 +837,30 @@ GPMSAFactory<V, M>::setUpEmulator()
     queso_assert_less_equal(SM_singularValues[i], SM_singularValues[i+1]);
   }
 
+  // Count the number of "nonzero" eigenvalues (eigenvalues over a tolerance)
+  double eigenvalue_tolerance = 1e-10;
+  for (unsigned int i = 0; i < SM_singularValues.sizeLocal(); i++) {
+    // All eigenvalues should be positive, so we don't take fabs
+    if (SM_singularValues[i] > eigenvalue_tolerance) {
```

Comment:
That tolerance ought to be relative, not absolute... but I guess we're rescaling data anyway so it doesn't matter much, and we're going to turn this into a user-controllable knob eventually so we can tweak it then.